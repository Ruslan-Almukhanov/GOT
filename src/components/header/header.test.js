import React from 'react';
import Header from './header'
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';

describe('testing <Header/>', () => {
    it('<Header renders corectly', () => {
        const header = render(<Router><Header /></Router>)
        expect(header).toMatchSnapshot();
    })
    it('logo rendered corectly', () => {
        const { getByText } = render(<Router><Header /></Router>)
        const logo = getByText(/Game of Thrones DB/)
        expect(logo).toBeTruthy()
    })
    it('menu rendered corectly', () => {
        const { getByTestId } = render(<Router><Header /></Router>)
        expect(getByTestId('test-menu')).toBeTruthy()
    })
})