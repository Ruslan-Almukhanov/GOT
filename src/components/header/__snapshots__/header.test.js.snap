// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <Header/> <Header renders corectly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdVaJa euyOib"
      >
        <h3
          class="sc-bwzfXH bmIlQj"
        >
          <a
            href="/"
          >
            Game of Thrones DB
          </a>
        </h3>
        <ul
          class="sc-htpNat jvnEHg"
          data-testid="test-menu"
        >
          <li>
            <a
              href="/characters/"
            >
              Characters
            </a>
          </li>
          <li>
            <a
              href="/houses/"
            >
              Houses
            </a>
          </li>
          <li>
            <a
              href="/books/"
            >
              Books
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdVaJa euyOib"
    >
      <h3
        class="sc-bwzfXH bmIlQj"
      >
        <a
          href="/"
        >
          Game of Thrones DB
        </a>
      </h3>
      <ul
        class="sc-htpNat jvnEHg"
        data-testid="test-menu"
      >
        <li>
          <a
            href="/characters/"
          >
            Characters
          </a>
        </li>
        <li>
          <a
            href="/houses/"
          >
            Houses
          </a>
        </li>
        <li>
          <a
            href="/books/"
          >
            Books
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
